libevm/triedb/firewood/firewood.go

```diff
@@ -0,0 +1,518 @@
+// Copyright 2025 the libevm authors.
+//
+// The libevm additions to go-ethereum are free software: you can redistribute
+// them and/or modify them under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation, either version 3 of the License,
+// or (at your option) any later version.
+//
+// The libevm additions are distributed in the hope that they will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser
+// General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see
+// <http://www.gnu.org/licenses/>.
+
+// The firewood package provides a [triedb.DBOverride] backed by [Firewood].
+//
+// [Firewood]: https://github.com/ava-labs/firewood
+package firewood
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/ava-labs/firewood-go-ethhash/ffi"
+
+	"github.com/ava-labs/libevm/common"
+	"github.com/ava-labs/libevm/core/rawdb"
+	"github.com/ava-labs/libevm/core/types"
+	"github.com/ava-labs/libevm/ethdb"
+	"github.com/ava-labs/libevm/libevm/stateconf"
+	"github.com/ava-labs/libevm/log"
+	"github.com/ava-labs/libevm/trie/trienode"
+	"github.com/ava-labs/libevm/trie/triestate"
+	"github.com/ava-labs/libevm/triedb"
+	"github.com/ava-labs/libevm/triedb/database"
+)
+
+const (
+	dbFileName  = "firewood.db"
+	logFileName = "firewood.log"
+)
+
+var _ triedb.DBOverride = (*Database)(nil)
+
+type Database struct {
+	// The underlying Firewood database, used for storing proposals and revisions.
+	// This must be exported so other packages (e.g. state sync) can access firewood-specific methods.
+	Firewood *ffi.Database
+
+	// proposalLock protects the proposal map and tree during updates.
+	proposalLock sync.RWMutex
+	// proposalMap provides O(1) access by state root to all proposals stored in the proposalTree
+	proposalMap map[common.Hash][]*proposal
+	// The proposal tree tracks the structure of the current proposals, and which proposals are children of which.
+	// This is used to ensure that we can dereference proposals correctly and commit the correct ones
+	// in the case of duplicate state roots.
+	// The root of the tree is stored here, and represents the top-most layer on disk.
+	proposalTree *proposal
+
+	// possibleProposals temporarily holds proposals created during a trie update.
+	// This is cleared after the update is complete and the proposals have been sent to the database.
+	possibleProposals []*proposal
+}
+
+type Config struct {
+	ChainDir             string
+	CleanCacheSize       int  // Size of the clean cache in bytes
+	FreeListCacheEntries uint // Number of free list entries to cache
```

Comment:
```suggestion
	CleanCacheSizeBytes       int 
	FreeListCacheEntries uint
```

The comments as they stood just reflected the code so were redundant and risk going out of date. Units should either be captured in the type (preferable) or at least in the field name, as these are self-documenting.